### PR TITLE
Ansible Module for Sending Rocket Chat Notifications

### PR DESCRIPTION
**Pull request by Deepakkothandan**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- Notification Moduel for Rocket Chat
##### ANSIBLE VERSION

```
 “2.0.2.0” 
```
##### SUMMARY

The module is used to send notification to Rocket Chat server via the Incoming Webhook Integration feature

```
```

### CODE REVIEW

File: notification/rocketchat.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          - title: "System A"
+            value: "load average: 0,74, 0,66, 0,63"
+            short: "true"
+          - title: "System B"
+            value: "load average: 5,16, 4,64, 2,43"
+            short: "true"
+
+"""
+
+ROCKETCHAT_INCOMING_WEBHOOK = 'https://%s/hooks/%s'
```
Comment (geekgonecrazy) [inline notification/rocketchat.py]: Might want to make the http or https configurable.  Most people should have https.  But they might not all.

**Comment by gregdek** [comment]
Thanks @Deepakkothandan for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @Deepakkothandan for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Deepakkothandan** [comment]
Resolved the build errors please change the PR status to  ready_for_review

**Comment by gundalow** [comment]
Note to self: gundalow_review_this

**Comment by gregdek** [comment]
Thanks @Deepakkothandan for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: notification/rocketchat.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    payload = {}
+    if color == "normal" and text is not None:
+        payload = dict(text=text)
+    elif text is not None:
+        payload = dict(attachments=[dict(text=text, color=color)])
+    if channel is not None:
+        if (channel[0] == '#') or (channel[0] == '@'):
+            payload['channel'] = channel
+        else:
+            payload['channel'] = '#'+channel
```
Comment (gundalow) [inline notification/rocketchat.py]: rocketchat.py:170:37: E226 missing whitespace around arithmetic operator

File: notification/rocketchat.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    description:
+      - Rocket Chat Incoming Webhook integration token.  This provides
+        authentication to Rocket Chat's Incoming webhook for posting
+        messages.
+    required: true
+  protocol:
+    description:
+      - Specify the protocol used to send notification messages before the webhook url. (i.e. http or https)
+    required: false
+    default: https
```
Comment (gundalow) [inline notification/rocketchat.py:47]: Please add `choices` to protocol

File: notification/rocketchat.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    required: false
+    default: None
+  username:
+    description:
+      - This is the sender of the message.
+    required: false
+    default: "Ansible"
+  icon_url:
+    description:
+      - Url for the message sender's icon (default C(http://www.ansible.com/favicon.ico))
```
Comment (gundalow) [inline notification/rocketchat.py]: URL

**Comment by gundalow** [comment]
This is Py3 compatible.

Just a few small things then this can be merged

needs_revision

**Comment by gregdek** [comment]
Thanks @Deepakkothandan for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Deepakkothandan** [comment]
@gundalow Thank you for reviewing the module. I have made the changes according to your comments and update the pull request, I have also rebased the commits. ready_for_review

**Comment by gregdek** [comment]
Thanks @Deepakkothandan for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Thanks for the quick turn around 

shipit

**Comment by gregdek** [comment]
Thanks again to @Deepakkothandan for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
failing build is unrelated. merging. Thx!

**Pull request merged**